core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -226,17 +226,14 @@ func (r *EvmRegistry) GetActiveUpkeepIDsByType(ctx context.Context, triggers ...
 	return keys, nil
 }
 
-// TODO: should be called with ocr2keepers.UpkeepPayload
-func (r *EvmRegistry) CheckUpkeep(ctx context.Context, mercuryEnabled bool, keys ...ocr2keepers.UpkeepKey) ([]ocr2keepers.UpkeepResult, error) {
+func (r *EvmRegistry) CheckUpkeeps(ctx context.Context /* mercuryEnabled bool, */, keys ...ocr2keepers.UpkeepPayload) ([]ocr2keepers.CheckResult, error) {
```

Comment:
I think this boolean is still useful? e.g. we will only set this to true for arbitrum.
I think we need to uncomment this and from the plugin side, pass this boolean.

Reply:
Can we run mercury individually per payload or do we make the assumption that all payloads will run mercury or not?

Reply:
synced offchain with Awbrey. @amirylm 
won't address this in this PR. we will pass this boolean via upkeep payload's configured upkeep field and keep this interface clean.